commands/user.go

```diff
@@ -131,283 +164,445 @@ func userGroup(b *bot.Bot) {
 			return messages.ErrIncorrectCmd(ctx.Command)
 		}
 
+		// Assigns current discord user.
+		// Returns error if no user is found.
 		user, err := b.Store.User(context.Background(), ctx.Event.Author.ID)
 		if err != nil {
 			return err
 		}
 
+		// Assigns name of crosspost group or pair.
 		name := ctx.Args.Get(0).Raw
+		// Assigns name of parent channel.
+		// Returns error if no channel exists.
 		parent := strings.Trim(ctx.Args.Get(1).Raw, "<#>")
 		if _, err := ctx.Session.Channel(parent); err != nil {
 			return messages.ErrChannelNotFound(err, parent)
 		}
 
+		// Assigns a crosspost channel when crosspost group is a pair.
+		// If group is not a pair leave empty.
+		children := []string{}
+		if ispair {
+			// Assigns channelID of crosspost channel.
+			child := strings.Trim(ctx.Args.Get(2).Raw, "<#>")
+
+			// Returns discord channel from channelID.
+			// If no discord channel is found return error.
+			ch, err := ctx.Session.Channel(child)
+			if err != nil {
+				return messages.ErrChannelNotFound(err, child)
+			}
+
+			// Checks if discord channel is a text channel.
+			if ch.Type != discordgo.ChannelTypeGuildText {
+				return messages.ErrIncorrectCmd(ctx.Command)
+			}
+
+			// Checks if crosspost channel is not parent channel.
+			if parent == child {
+				return messages.ErrIncorrectCmd(ctx.Command)
+			}
+
+			// Checks if crosspost channel is not parent of an existing group.
+			if _, ok := user.FindGroupByParent(child); ok {
+				return messages.ErrInsertGroup(name, child)
+			}
+
+			// Appends crosspost channel to children.
+			children = append(children, child)
+		}
+
+		// Creates crosspost group when checks suffice.
 		_, err = b.Store.CreateCrosspostGroup(context.Background(), user.ID, &store.Group{
 			Name:     name,
 			Parent:   parent,
-			Children: []string{},
+			Children: children,
+			IsPair:   ispair,
 		})
 
+		// Returns error if any mongoDB error is raised.
 		if err != nil {
-			switch {
-			case errors.Is(err, mongo.ErrNoDocuments):
-				return messages.ErrInsertGroup(name, parent)
-			default:
+			if err = errMongoDB(err, messages.ErrInsertGroup(name, parent)); err != nil {
 				return err
 			}
 		}
 
-		eb := embeds.NewBuilder()
-
-		msg := fmt.Sprintf("Created a group `%v` with parent channel <#%v> | `%v`", name, parent, parent)
-		eb.SuccessTemplate(msg)
+		// Assign success message for crosspost group or pair.
+		var msg string
+		if ispair {
+			msg = messages.UserCreatePairSuccess(name, parent, children[0])
+		} else {
+			msg = messages.UserCreateGroupSuccess(name, parent)
+		}
 
-		return ctx.ReplyEmbed(eb.Finalize())
+		// Returns success message embed.
+		return successMessage(ctx, msg)
 	}
 }
 
+// delgroup deletes a crosspost group.
 func delgroup(b *bot.Bot) func(ctx *gumi.Ctx) error {
 	return func(ctx *gumi.Ctx) error {
+		// Checks if Boe Tea command is correct.
 		if ctx.Args.Len() < 1 {
 			return messages.ErrIncorrectCmd(ctx.Command)
 		}
 
+		// Assigns current discord user.
```

Comment:
Some comments are also just misleading, the user isn't assigned it's retrieved from the database.